pkg/ddl/000031_commits_parents_index.up.sql

```diff
@@ -1,4 +1,22 @@
 BEGIN;
-CREATE INDEX graveler_commits_idx_parents ON graveler_commits(parents);
+
 CREATE INDEX graveler_branches_idx_commit_id ON graveler_branches(commit_id);
+
+-- function to return the first parent:
+-- this takes into consideration commits created before fixing the parent switch bug (https://github.com/treeverse/lakeFS/issues/1730)
+CREATE OR REPLACE FUNCTION first_parent(parents text[], version int)
+RETURNS text AS
+    $$
+    DECLARE
+        parent_switch_commit_version int;
+    BEGIN
+        parent_switch_commit_version := 1;
```

Comment:
Does this work?  I think it will be more readable.  (You might also drop the `CONSTANT` if that's from too advanced a Postgres version, it will still be easier for me to read.)  https://www.postgresql.org/docs/13/plpgsql-declarations.html says it should work, even in version 11, so I reckon it might.

```suggestion
    DECLARE
        parent_switch_commit_version CONSTANT INT = 1;
    BEGIN
```